snapcraft/internal/lifecycle.py

```diff
@@ -252,6 +253,14 @@ def _create_meta(self, step, part_names):
             meta.create_snap_packaging(self.config.data,
                                        self.project_options)
 
+    def _record_snapcraft(self, step, part_names):
```

Comment:
we would also be splitting logic and ownership if it is not in meta; the `snap` directory logic would be split up otherwise